vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/FSRegisteredPackage.java

```diff
@@ -39,24 +39,27 @@
      */
     private static final Logger log = LoggerFactory.getLogger(FSPackageRegistry.class);
 
-    private VaultPackage vltPkg;
+    private FSInstallState installState;
     private FSPackageRegistry registry;
 
-    public FSRegisteredPackage(FSPackageRegistry registry, VaultPackage vltPkg) throws IOException, RepositoryException {
-        this.vltPkg = vltPkg;
+    private VaultPackage vltPkg = null;
+
+    public FSRegisteredPackage(FSPackageRegistry registry, FSInstallState installState) throws IOException, RepositoryException {
+        this.installState = installState;
```

Comment:
what about the other methods like `isInstalled()` or `getInstallationTime()`. shouldn't they rely on the immutable installstate?

otherwise, I would not store the installState in this object, but only remember `packageId` and `filePath`. i..e:
```
this.packageId = installState.getPackageId();
this.filePath = installState.getFilePath();
```